app/src/pages/serverErrorPage/ServerErrorPage.test.tsx

```diff
@@ -0,0 +1,86 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import ServerErrorPage from './ServerErrorPage';
+import { act } from 'react-dom/test-utils';
+import userEvent from '@testing-library/user-event';
+
+const mockedUseNavigate = jest.fn();
+
+jest.mock('react-router', () => ({
+    useNavigate: () => mockedUseNavigate,
+    useLocation: () => jest.fn(),
+}));
+
+describe('ServerErrorPage', () => {
+    beforeEach(() => {
+        process.env.REACT_APP_ENVIRONMENT = 'jest';
+    });
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    describe('Rendering', () => {
+        it('renders page content with default error message and id', () => {
+            render(<ServerErrorPage />);
+
+            expect(
+                screen.getByRole('heading', {
+                    name: 'Sorry, there is a problem with the service',
+                }),
+            ).toBeInTheDocument();
+            expect(screen.getByText('An unknown error has occurred.')).toBeInTheDocument();
+            expect(
+                screen.getByText(
+                    "Try again by returning to the previous page. You'll need to enter any information you submitted again.",
+                ),
+            ).toBeInTheDocument();
+            expect(
+                screen.getByRole('button', {
+                    name: 'Return to previous page',
+                }),
+            ).toBeInTheDocument();
+            expect(
+                screen.getByRole('heading', {
+                    name: 'If this error keeps appearing',
+                }),
+            ).toBeInTheDocument();
+            expect(
+                screen.getByRole('link', {
+                    name: 'Contact the NHS National Service Desk',
+                }),
+            ).toBeInTheDocument();
+            expect(screen.getByText('UNKNOWN_ERROR')).toBeInTheDocument();
+        });
+
+        it('renders page content with error message and id', () => {
+            jest.spyOn(URLSearchParams.prototype, 'get').mockReturnValue('CDR_5001');
+            render(<ServerErrorPage />);
+
+            expect(
+                screen.getByRole('heading', {
+                    name: 'Sorry, there is a problem with the service',
+                }),
+            ).toBeInTheDocument();
+            expect(screen.getByText('Internal error')).toBeInTheDocument();
+            expect(screen.queryByText('An unknown error has occurred.')).not.toBeInTheDocument();
+            expect(screen.getByText('CDR_5001')).toBeInTheDocument();
+            expect(screen.queryByText('UNKNOWN_ERROR')).not.toBeInTheDocument();
+        });
```

Comment:
another test case for invalid error code. does the page break? does it render default content?